### DISCUSSION TITLE
Feature Request: Read Only Config Changeable through UI

### DISCUSSION THREAD

**Discussion by minmaxat** [dani-garcia/vaultwarden#4180] [Ideas]
Read Only Config Changeable through UI -> this would be great for other Installations like Home Assistant. 
I think it is no Problem that the Changes Work only after a Restart.

**Comment by minmaxat** [comment]
Especially i think about the Push Notification Key which will be usefull for most installations

**Reply by tessus** [reply]
there was another discussion about this as well: #4164

**Comment by BlackDex** [comment]
I have been thinking about this. And there might be a way.
I just hope it doesn't cause false/invalid issue reports of users who forgot to restart the instance.

**Comment by BlackDex** [comment]
Also, moving this to the idea's discussions.

**Comment by minmaxat** [comment]
Thanks for looking into. Maybe we can put there a red Flag for this Settings which are not restartet yet -> and, i think, especially the Home Assistant Users are used to restart some things when they change something. 
There already some Topics in the Hassio-Addons Repo which link to this: 
ie.: https://github.com/hassio-addons/addon-bitwarden/issues/283

**Comment by Getslow6** [comment]
As mentioned by @BlackDex  in the #4164 discussion, it might be able to edit the config.json file (although not officially supported or recommended).

I haven't tried it yet because I don't want to try it on my 'production instance', but using the using the 'Advanced SSH & Web terminal' addon you could edit the config.json file using the following steps;
```
sudo docker exec -ti addon_a0d7b954_bitwarden bash
cd data
echo "$(jq '. += {"push_enabled": true, "push_installation_id": "YOUR_INSTALLATION_ID", "push_installation_key": "YOUR_INSTALLATION_KEY"}' config.json)" > config.json
```
And then restart.

**Reply by stefan0xC** [reply]
The problem with manually editing the `config.json` is that it's easily overwritten. But since it does work I've clarified the issue on the [wiki page](https://github.com/dani-garcia/vaultwarden/wiki/Configuration-overview#using-the-admin-page).

**Reply by clementko** [reply]
I can echo that it does work. I manually add push_enabled: ture, id and key into config.json using the "Advanced SSH & Web Terminal"
